ibis/client.py

```diff
@@ -243,7 +234,8 @@ def execute(self, expr, params=None, limit='default', async=False):
 
     def _execute_query(self, ddl, async=False):
         klass = self.async_query if async else self.sync_query
-        return klass(self, ddl).execute()
+        inst = klass(self, ddl)
+        return inst.execute()
```

Comment:
The previous one was nicer :)

Reply:
Yeah, but this is +epsilon easier to debug.